File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) {
        //find the max one and also the freq of the elements
        int maxi = *max_element(piles.begin(), piles.end());
        int n = piles.size();
        vector<int> freq(maxi + 1, 0);

        for(int i = 0; i < n; i++) {
            freq[piles[i]]++;
        }  

        int maxCoins = 0;
        int chance = n / 3;
        bool turn = true;

        while(chance) {
            if(freq[maxi] > 0) {
                if(turn) turn = false;
                else {
                    chance--;
                    turn = true;
                    maxCoins += maxi;
                }
                freq[maxi]--;
            }
            else {
                maxi--;
            }
        }

        return maxCoins;
    }
};

File: 1683-maximum-number-of-coins-you-can-get/maximum-number-of-coins-you-can-get.py
class Solution:
    def maxCoins(self, piles: List[int]) -> int:
        piles.sort()
        maxSum = 0
        n = len(piles)

        print(piles)
        for i in range(n // 3, len(piles), 2):
            print(i)
            maxSum += piles[i]
        
        return maxSum